src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -870,6 +894,34 @@ ExecuteSingleTask(QueryDesc *queryDesc, Task *task,
 
 
 /*
+ * GetPlacementConnectionsReadyForTwoPhaseCommit iterates over the task placement list,
+ * get connections and mark them as critical. Finally, establishes connections to all
+ * placements at once.
+ */
+static void
+GetPlacementConnectionsReadyForTwoPhaseCommit(List *taskPlacementList)
+{
+	ListCell *taskPlacementCell = NULL;
+	List *multiConnectionList = NIL;
+
+	foreach(taskPlacementCell, taskPlacementList)
+	{
+		ShardPlacement *taskPlacement = (ShardPlacement *) lfirst(taskPlacementCell);
+		int connectionFlags = SESSION_LIFESPAN;
+		MultiConnection *multiConnection = GetNodeConnection(connectionFlags,
```

Comment:
Oh, I've just read the related function comments, they already explain this.

I'll update accordingly.